src/Components/Aspire.Azure.Messaging.ServiceBus/AzureMessagingServiceBusSettings.cs

```diff
@@ -114,9 +130,36 @@ void IConnectionStringSettings.ParseConnectionString(string? connectionString)
                 return;
             }
 
-            // if we got here, it's a full connection string
-            // use the original connection string since connectionBuilder was modified above.
-            ConnectionString = connectionString;
+            // Use the connection string with the entitypath removed to
+            // support configuring processors/receivers/etc from the service
+            // bus client with names derived from settings.
+            ConnectionString = connectionBuilder.ConnectionString;
```

Comment:
This surprises me a little bit. You used a connection that had one Queue on it, but want to override it with a new Queue. I guess this is consistent with what we do elsewhere - like the new CosmosDB change. I am a little concerned that we aren't using the original connection string.

Is there any change to the existing code if you just call `AddAzureServiceBusClient` and the connection string had an EntityPath before, but now it doesn't?

Reply:
Yes, this bit of code is designed to workaround the [ValidateEntityName](https://github.com/Azure/azure-sdk-for-net/blob/ecc9cd50619413f39f99949efd03e5d0fa89b7a4/sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClient.cs#L722) implementation in the SDK that assumes the QueueOrTopicName and SubscriptionName must match those in the connection string. By removing the entity path from the connection string, we circumvent the name checks it does.

> Is there any change to the existing code if you just call AddAzureServiceBusClient and the connection string had an EntityPath before, but now it doesn't?

I guess one breaking change here is that you are no longer getting the validation on teh entity names for any subclients that you were generating.